src/components/ListView.js

```diff
@@ -1,28 +1,29 @@
-import React, { useEffect, useState } from 'react';
-import { db } from '../lib/firebase';
-import { v4 as uuidv4 } from 'uuid';
+import React from 'react';
 
-const ListView = ({ token }) => {
-  const [purchaseItemCollection, setpurchaseItemCollection] = useState([]);
+import { v4 as uuidv4 } from 'uuid';
 
-  useEffect(() => {
-    db.collection('shoppinglist')
-      .where('token', '==', token)
-      .get()
-      .then((querySnapshot) => {
-        querySnapshot.forEach((doc) => {
-          setpurchaseItemCollection(doc.data().items);
-        });
-      });
-  });
+import ListItem from '../components/ListItem';
 
+const ListView = ({ shoppingList, loading, error }) => {
   return (
-    <ul className="lists">
-      {purchaseItemCollection &&
-        purchaseItemCollection.map((item) => (
-          <li key={uuidv4()}>{item.name}</li>
-        ))}
-    </ul>
+    <>
+      {error && <strong>Error: {JSON.stringify(error)}</strong>}
+      {loading && (
+        <div className="bouncing-loader">
+          <div></div>
+          <div></div>
+          <div></div>
+        </div>
+      )}
+      <ul className="list">
+        {shoppingList &&
+          shoppingList.map((list) =>
+            list.items
+              .map((item) => item)
```

Comment:
Why is this extra map added? It just seems to go through all the items and return them?

Reply:
Good catch, I've missed that